File: kalitech-engine/src/main/resources/kalitech/engine/modules/Entity/Entity.js
"use strict";

const {req} = require("./helpers/EntUtil.js");
const {EntApi} = require("./helpers/EntApi.js");

function create(engine, K) {
    req(engine, "[ENT] engine is required");

    const api = new EntApi(engine, K);

    return Object.freeze({
        create: api.create.bind(api),

        $: api.$.bind(api),
        player$: api.player$.bind(api),
        capsule$: api.capsule$.bind(api),
        box$: api.box$.bind(api),
        sphere$: api.sphere$.bind(api),

        preset: api.preset.bind(api),
        bodyDefaults: api.bodyDefaults.bind(api),
        presets: api.presets.bind(api),

        idOf: api.idOf.bind(api),
        uuidOf: api.uuidOf.bind(api)
    });
}

create.META = {
    moduleId: "entity",
    globalName: "ENT",
    version: "2.0.2",
    description: "Declarative entity builder (UUID-only). Returns {core, handle} (compat). Core mirrors Java snapshot/components.",
    engineMin: "0.2.0"
};

module.exports = create;
module.exports.META = create.META;

File: kalitech-engine/src/main/resources/kalitech/engine/modules/Entity/Entity.lua
local M = {}
local luaRuntime = require("@builtin/lua_runtime")
local Classes = luaRuntime.class
local lua_require_result_0 = require("./helpers/EntUtil.lua")
req = lua_require_result_0.req
local lua_require_result_1 = require("./helpers/EntApi.lua")
EntApi = lua_require_result_1.EntApi
create = setmetatable(
    {},
    {__call = function(lua_, self, engine, K)
        req(_G, engine, "[ENT] engine is required")
        local api = Classes:construct(EntApi, engine, K)
        return KObject:freeze({
            create = KFunction:bind(api.create, api),
            ["$"] = KFunction:bind(api["$"], api),
            ["player$"] = KFunction:bind(api["player$"], api),
            ["capsule$"] = KFunction:bind(api["capsule$"], api),
            ["box$"] = KFunction:bind(api["box$"], api),
            ["sphere$"] = KFunction:bind(api["sphere$"], api),
            preset = KFunction:bind(api.preset, api),
            bodyDefaults = KFunction:bind(api.bodyDefaults, api),
            presets = KFunction:bind(api.presets, api),
            idOf = KFunction:bind(api.idOf, api),
            uuidOf = KFunction:bind(api.uuidOf, api)
        })
    end}
)
create.META = {
    moduleId = "entity",
    version = "2.0.2",
    description = "Declarative entity builder (UUID-only). Returns {core, handle}. Core mirrors Java snapshot/components.",
    engineMin = "0.2.0"
}
M = create
M.META = create.META

return M
